public/pages/workflows/new_workflow/new_workflow.test.tsx

```diff
@@ -31,8 +47,49 @@ const renderWithRouter = () =>
   );
 
 describe('NewWorkflow', () => {
-  test('renders the search bar', () => {
-    const { getByPlaceholderText } = renderWithRouter();
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(ReactReduxHooks, 'useAppDispatch').mockReturnValue(mockDispatch);
+  });
+
+  test('renders the preset workflow templates', () => {
+    const { getByPlaceholderText, getAllByText } = renderWithRouter();
     expect(getByPlaceholderText('Search')).toBeInTheDocument();
+    expect(getAllByText('Custom')).toHaveLength(1);
+    expect(getAllByText('Hybrid Search')).toHaveLength(1);
+    expect(getAllByText('Multimodal Search')).toHaveLength(1);
+    expect(getAllByText('Semantic Search')).toHaveLength(1);
+    expect(getAllByText('Retrieval-Augmented Generation')).toHaveLength(1);
+    expect(getAllByText('Sentiment Analysis')).toHaveLength(1);
```

Comment:
for consistency, let's leverage the `WORKFLOW_TYPE` enum. In fact, we should be able to iterate through them in one loop (besides `WORKFLOW_TYPE.UNKNOWN`). Can do something like:

```
Object.values(WORKFLOW_TYPE).forEach((type) => {
  if (type !== WORKFLOW_TYPE.UNKNOWN) {
     expect(getAllByText(type)).toHaveLength(1)
  }
})
```

Reply:
Then as we add more, they will be covered in this test case.